### PR TITLE
fix(skeleton-components): update labels & attributes for Checkbox, NumberInput, Select, Slider, TextArea, and TextInput

### LINKED ISSUE #4176: AVT 1 - Skeleton: Checkbox, NumberInput, Select, Slider, TextArea & TextInput DAP violation for missing label
## Environment
macOS Mojave version 10.14.6
Chrome Version Version 77.0.3865.90
Carbon v10 - React
DAP IBM Accessibility 1.8.0.0 Sept 2019 Ruleset

Detailed Description
Go to the React skeleton for Checkbox, NumberInput, Select, Slider, TextArea & TextInput
Run DAP and the following issue is found on the on each skeleton component (see screenshots):
checkbox
numberinput
select
TextArea
TextInput

### PR DESCRIPTION
**Pull request by jendowns**
Closes #4176
Also partially addresses #4310 

The components flagged in #4176 had `label` elements that weren't tied to any particular `input`, because these are all ~non-interactive~ skeleton components. So I changed those `label` elements to `spans`

#### Changelog

**Changed**

- For each component skeleton, I changed the `label` element to a `span`
- For each component's skeleton story, I added `aria-label="loading [component name]"`, `aria-live="assertive"`, `role="status"`, and `tabindex="0"` to the wrapper element

#### Testing / Reviewing

Open the `carbon-components-react` deployment review and check these skeleton components against the latest Sept 2019 ruleset with the DAP tool.

### CODE REVIEW

**Comment by jendowns** [comment]
Tagging @dakahn for this one as well ☝️

**Comment by dakahn** [comment]
Our requirements around skeletons (loading states) for components are changing. Referencing two things here:
https://www.w3.org/WAI/WCAG21/Understanding/status-messages.html
https://www.ibm.com/able/guidelines/ci162/status_messages_71.html

There's a lot in there, but relevant to this PR is our skeletons do actually need to be interactable in so far as they can receive focus (be tabbable) and communicate the status of that particular component.

**Review by dakahn — COMMENTED** [review]
Our requirements around skeletons (loading states) for components are changing. Referencing two things here:
https://www.w3.org/WAI/WCAG21/Understanding/status-messages.html
https://www.ibm.com/able/guidelines/ci162/status_messages_71.html

There's a lot in there, but relevant to this PR is our skeletons do actually need to be interactable in so far as they can receive focus (be tabbable) and communicate the status via aria of that particular component.

**Comment by jendowns** [comment]
Thanks @dakahn that's very interesting 🤔 I could add a tabindex to these elements for now?

**Comment by jendowns** [comment]
Thanks again @dakahn -- I just added `tabindex='0'`, `role='status'`, `aria-live="assertive"` to each of these skeleton components in my latest commit 👍

**Review by asudoh — APPROVED** [review]
LGTM 👍 - Thanks @jendowns!

File: packages/react/src/components/TextInput/TextInput.Skeleton.js [hunk 1]
```diff
@@ -11,19 +11,17 @@ import { settings } from 'carbon-components';
...
-  return (
-    <div className={`${prefix}--form-item`}>
-      {label}
-      <div className={`${prefix}--skeleton ${prefix}--text-input`} />
-    </div>
-  );
-};
+const TextInputSkeleton = ({ hideLabel }) => (
+  <div className={`${prefix}--form-item`}>
+    {hideLabel && <span className={`${prefix}--label ${prefix}--skeleton`} />}
```
Comment (emyarod) [inline packages/react/src/components/TextInput/TextInput.Skeleton.js]:
```suggestion
    {!hideLabel && <span className={`${prefix}--label ${prefix}--skeleton`} />}
```

**Review by emyarod — COMMENTED** [review]
`hideLabel` logic for the `TextInput` skeleton is reversed, otherwise looks good to me

**Comment by jendowns** [comment]
Thanks @emyarod! Just committed your suggestion 👍

**Review by jnm2377 — APPROVED** [review]
LGTM! I'm not seeing the DAP violation anymore 👍

**Comment by joshblack** [comment]
cc @dakahn does this align with our skeleton work that we need to do? Wanted to double-check before merging, seems like `aria-live="assertive"` is being used and wasn't sure if we wanted that, especially if we don't have any `aria-label` or message to convey

**Comment by dakahn** [comment]
@joshblack Yeah, I worked with Jen a little on the requirements for this one. The `aria-live` attribute should be communicating that whatever component they're interacting with is in a state of loading and ideally when it's loaded. This is in line with the changes we have cued up for the rest of the component's loading states. We're tracking that work here: https://github.com/carbon-design-system/carbon/issues/4310

**Comment by joshblack** [comment]
@dakahn should these skeletons include the `aria-label` then? As an aside, should components even be the ones defining these or should we instead provide a primitive for folks to specify that a region of a page is loading? Thinking about the experience, it seems bizarre that we'd announce that a component is loading as this has no meaning to an end-user.

**Comment by dakahn** [comment]
The meaning they're conveying is "Hey, you can't interact with this form item yet" or whatever. 

Not sure about that application level primitive, but our distinct loading state components are required to meet status message criteria found here
https://www.w3.org/WAI/WCAG21/Understanding/status-messages.html
and here
https://www.ibm.com/able/guidelines/ci162/status_messages_71.html

**Review by emyarod — APPROVED** [review]
thanks for making the change

**Comment by jendowns** [comment]
@joshblack @asudoh @dakahn @emyarod @jnm2377  I just pushed a commit that adds `aria-label` attributes to these skeleton components, per #4310

**Review by dakahn — APPROVED** [review]

**Comment by joshblack** [comment]
@dakahn I would think we'd want folks to specify that a region is loading versus specific components, or at the very least incorporate `useAnnouncer` such that the messages are delayed. If we had 5 input elements in a form that is loading, wouldn't only the last one announce completely as the other ones would be interrupted by the next message in the source order? Also, what happens once the part of the page is loaded? Should there be another aria-live message communicated?

**Comment by joshblack** [comment]
Quick demo of what I mean: 

[Screen Recording 2019-10-15 at 11.55.29 AM.zip](https://github.com/carbon-design-system/carbon/files/3730529/Screen.Recording.2019-10-15.at.11.55.29.AM.zip)

Codepen: https://codepen.io/joshblack/pen/ZEEWZYr?editors=1011

The way it works is that clicking the button would change the attributes of each of the outlined spans so that `aria-live` is set and an appropriate aria label is set. You almost don't even realize in VO that multiple things have `aria-live` as they are never announced

**Comment by jendowns** [comment]
@dakahn @joshblack Oh whoa, that's interesting... so maybe I should remove `aria-live` from these components for now until this is investigated further?

**Comment by joshblack** [comment]
@jendowns yeah, my guess is that if we want to remove the violations we should just treat these as decorative content until we figure out a live region example. This would mean removing those aria attributes, I believe, in the skeletons.

The only follow-up would be that the stories should most likely be updated to be in a live region example that does have these attributes, right @dakahn? I think if we can figure that out then this should be good to go

**Comment by jendowns** [comment]
@dakahn @joshblack I just removed the `aria-live` attributes from these skeleton components. Please let me know if you want me to remove the `aria-label` attributes as well.

**Comment by joshblack** [comment]
I say go for it, what do you think @dakahn?

Only thing will be to make sure the skeleton stories do include a region that has all of these attributes on it.

**Comment by jendowns** [comment]
@dakahn @joshblack in my last 2 comments, I removed `aria-label` from the skeleton components themselves, and then I added `aria-label="loading [component name]"` and `aria-live="assertive"` to each of these skeleton component's wrappers in storybook. Is that the desired direction here?

**Comment by dakahn** [comment]
Hmm, tested with NVDA on latest Chrome and I'm not getting an announcement on page load that there is a component loading. I think your aria might be getting clobbered by the aria on the "main" Storybook element, but I'm not sure.

**Comment by joshblack** [comment]
@dakahn hmm, seems to work in VO so maybe there is an additional requirement for NVDA that we don't know about?

**Comment by dakahn** [comment]
That's fine. Green check from me. Let's get it in.

**Comment by joshblack** [comment]
@dakahn @jendowns do we still need role and tabindex on the skeletons? or would these move to the regions?

**Comment by dakahn** [comment]
Just to clarify what we're looking for here:

We want no aria on the skeletons themselves. This will clear the DAP violation. Then we want a wrapper div with the aria required to announce the loading state. This doesn't technically help our shipping skeletons be any more accessible, but the plan is to work on a region pattern users can implement _around_ their skeletons in a coming work cycle.

**Comment by jendowns** [comment]
@dakahn @joshblack I just pushed a commit that moves `role="status"` and `tabindex="0"` from the skeleton component's code to their wrapper elements in storybook.

**Review by joshblack — APPROVED** [review]

**Pull request merged**